plugins/deep-link/src/lib.rs

```diff
@@ -105,10 +154,164 @@ mod imp {
     }
 
     impl<R: Runtime> DeepLink<R> {
-        /// Get the current URLs that triggered the deep link.
+        /// Get the current URLs that triggered the deep link. Use this on app load to check whether your app was started via a deep link.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **Windows / Linux**: Unsupported.
         pub fn get_current(&self) -> crate::Result<Option<Vec<url::Url>>> {
             Ok(self.current.lock().unwrap().clone())
         }
+
+        /// Register the app as the default handler for the specified protocol.
+        ///
+        /// - `protocol`: The name of the protocol without `://`. For example, if you want your app to handle `tauri://` links, call this method with `tauri` as the protocol.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **macOS / Android / iOS**: Unsupported.
+        pub fn register<S: AsRef<str>>(&self, _protocol: S) -> crate::Result<()> {
+            #[cfg(windows)]
+            {
+                let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+                let base = Path::new("Software")
+                    .join("Classes")
+                    .join(_protocol.as_ref());
+
+                let exe = tauri::utils::platform::current_exe()?
+                    .display()
+                    .to_string()
+                    .replace("\\\\?\\", "");
+
+                let (key, _) = hkcu.create_subkey(&base)?;
+                key.set_value(
+                    "",
+                    &format!("URL:{} protocol", self.app.config().identifier),
+                )?;
+                key.set_value("URL Protocol", &"")?;
+
+                let (icon, _) = hkcu.create_subkey(base.join("DefaultIcon"))?;
+                icon.set_value("", &format!("{},0", &exe))?;
+
+                let (cmd, _) =
+                    hkcu.create_subkey(base.join("shell").join("open").join("command"))?;
+
+                cmd.set_value("", &format!("{} \"%1\"", &exe))?;
+            }
+
+            #[cfg(target_os = "linux")]
+            {
+                let bin = tauri::utils::platform::current_exe()?;
+                // TODO: Research if this conflicts with appimage integrations somehow. If it does, re-add a `-handler` suffix if running in an appimage.
```

Comment:
is this TODO resolved?

Reply:
no

Reply:
i added back the -handler suffix. afaik this is how you're (kinda) supposed to do it, even if it breaks pop os 🤷 

if someone else wanna research this more deeply then they can go ahead, but for me this is good enough.